tst/EnergyPlus/unit/WeatherManager.unit.cc

```diff
@@ -678,3 +682,133 @@ TEST_F(EnergyPlusFixture, WeatherManager_NoLocation) {
+        "  Enthalpy,                               !- Humidity Condition Type",
+        "  ,                                       !- Wetbulb or DewPoint at Maximum Dry-Bulb {C}",
+        "  ,                                       !- Humidity Condition Day Schedule Name",
+        "  ,                                       !- Humidity Ratio at Maximum Dry-Bulb {kgWater/kgDryAir}",
+        "  90500.0,                                !- Enthalpy at Maximum Dry-Bulb {J/kg}",
+        "  ,                                       !- Daily Wet-Bulb Temperature Range {deltaC}",
+        "  100511.,                                !- Barometric Pressure {Pa}",
+        "  3.2,                                    !- Wind Speed {m/s}",
+        "  220,                                    !- Wind Direction {deg}",
+        "  No,                                     !- Rain Indicator",
+        "  No,                                     !- Snow Indicator",
+        "  No,                                     !- Daylight Saving Time Indicator",
+        "  ASHRAETau,                              !- Solar Model Indicator",
+        "  ,                                       !- Beam Solar Day Schedule Name",
+        "  ,                                       !- Diffuse Solar Day Schedule Name",
+        "  0.773,                                  !- ASHRAE Clear Sky Optical Depth for Beam Irradiance (taub) {dimensionless}",
+        "  1.428;                                  !- ASHRAE Clear Sky Optical Depth for Diffuse Irradiance (taud) {dimensionless}",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    SimulationManager::OpenOutputFiles();
+    // reset eio stream
+    compare_eio_stream("", true);
+
+    bool ErrorsFound(false);
+    DataEnvironment::TotDesDays = 1;
+    // setup environment state
+    Environment.allocate(DataEnvironment::TotDesDays);
+    DesignDay.allocate(DataEnvironment::TotDesDays);
+
+    Environment(1).DesignDayNum = 1;
+    DataGlobals::MinutesPerTimeStep = 60;
+    DataGlobals::NumOfTimeStepInHour = 1;
+    DataGlobals::BeginSimFlag = true;
+    DataReportingFlags::DoWeatherInitReporting = true;
+
+    WeatherManager::SetupInterpolationValues();
+    WeatherManager::AllocateWeatherData();
+
+    WeatherManager::GetDesignDayData(DataEnvironment::TotDesDays, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+
+
+    WeatherManager::SetUpDesignDay(1);
+    EXPECT_EQ(WeatherManager::DesDayInput(1).HumIndType, DDHumIndType_Enthalpy);
+    EXPECT_EQ(WeatherManager::DesDayInput(1).HumIndValue, 90500.0);
+
+    unsigned n_RH_not100 = 0;
+    for (int Hour = 1; Hour <= 24; ++Hour) {
+        for (int TS = 1; TS <= DataGlobals::NumOfTimeStepInHour; ++TS) {
+            EXPECT_GE(WeatherManager::TomorrowOutRelHum(TS, Hour), 0.);
+            EXPECT_LE(WeatherManager::TomorrowOutRelHum(TS, Hour), 100.);
+            if (WeatherManager::TomorrowOutRelHum(TS, Hour) < 100.) {
+                ++n_RH_not100;
+            }
+        }
+    }
```

Comment:
Reproduces defect file condition. By multiplying (wrongly) J/kg by 1000 before, it was giving always 100% RH.